### ISSUE TITLE
Noclip Interferes with Buttons and Switches

### ISSUE THREAD

**Issue by ForLoveOfCats** [JimmyCushnie/TUNG#36]
**OS:** Manjaro GNU/Linux x64 w/ Plasma 5
**TUNG Version:** 0.2.4
**Mods or Addons:** None

With noclip enabled the user cannot interact with standard buttons or switches. Nothing else is affected.

List of **affected** items:
* Standard Buttons
* Standard Switch

List of **unaffected** items:
* Panel Button
* Panel Switch
* Display
* Panel Display
* Noise Maker
* Label
* Panel Label

Cheers!

**Comment by JimmyCushnie** [comment]
Unable to reproduce.
https://www.youtube.com/watch?v=vtSjpV59ejY

**Comment by ForLoveOfCats** [comment]
That is what I intended to convey. Only **non**-panel buttons and switches are affected by this bug.

**Comment by JimmyCushnie** [comment]
In the video, I am interacting with non-panel buttons and switches with noclip enabled. Isn't this the bug? Or am I misunderstanding you?

**Comment by ForLoveOfCats** [comment]
Woops, my monkey brain got confused by the fact that they are on a vertical board.

**Comment by JimmyCushnie** [comment]
No worries haha. Are you still getting the issue? Maybe it's a rare occurance

**Comment by ForLoveOfCats** [comment]
I cannot currently test due to #35

**Comment by JimmyCushnie** [comment]
Okay, so I figured out this bug. When you interact with something, the game casts a ray from the camera into the world. Usually, the player is exempt from the list of things the ray passes through, but I forgot to remove the noclip player from that list. However, due to the shape of the player, the ray only hits it if you're looking fairly down or up, so I was able to do it just fine when I was looking straight ahead.

This bug is fixed along with a few other noclip related bugs for 0.2.6.
